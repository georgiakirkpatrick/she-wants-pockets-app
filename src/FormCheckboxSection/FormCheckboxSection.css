.empty {
    height: 0;
}

.FormCheckboxSection {
    width: 100%;
    padding: 5px;
    margin: 2px 0;
    background-color: rgb(238, 238, 238);
}

.FormCheckboxSection p {
   font-size: 16px;
   font-weight: 500;
   padding: 10px;
}

.FormCheckboxSection__option {
    padding: 10px;
}

.FormCheckboxSection__option label {
    padding: 0 10px;
}

.FormCheckboxSection__options-container {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
}

.FormCheckboxSection__options-container .categories {
    height: 40vh
}

.FormCheckboxSection__options-container .fibers {
    height: 45vh
}

.FormCheckboxSection__options-container .sizes {
    height: 55vh;
}