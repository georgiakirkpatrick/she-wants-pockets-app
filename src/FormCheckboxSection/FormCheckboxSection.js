import React from 'react'
import './FormCheckboxSection.css'

const FormCheckboxSection = props => {
    const selectedOptions = props.selectedOptions 
        ? props.selectedOptions 
        : Object.fromEntries(props.options.map((o) => [o.id, false]))

    const prompt = props.prompt 
        ? <p>{props.prompt}</p> 
        : <div className='empty' />
        
    const options = props.options.map(option => {
        return (
            <div key={option.id} className='FormCheckboxSection__option'>
                <input type='checkbox' 
                    id={option.id} 
                    name={option.name ? option.name : option.id} 
                    checked={selectedOptions[option.id]}
                    onChange={props.handleChange}>
                </input>
                <label htmlFor={option.id}>{option.text}</label>
            </div>
        )
    })

    return (
        <div className='FormCheckboxSection'>
            {prompt}
            <div className='FormCheckboxSection__options-container'>
                {options}
            </div>
        </div>
    )
}

FormCheckboxSection.defaultProps = {
    selectedOptions: [],
    options: [],
    prompt: '',
    change: () => {}
}

export default FormCheckboxSection