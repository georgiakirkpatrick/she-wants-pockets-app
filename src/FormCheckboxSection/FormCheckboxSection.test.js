import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FormCheckboxSection from './FormCheckboxSection';

describe('FormCheckboxSection', () => {
        it ('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<FormCheckboxSection />, div);
            ReactDOM.unmountComponentAtNode(div);
        });
    
        it ('render the UI as expected', () => {
            const tree = renderer
                .create(<FormCheckboxSection />)
                .toJSON();
            expect(tree).toMatchSnapshot()
        });
    });