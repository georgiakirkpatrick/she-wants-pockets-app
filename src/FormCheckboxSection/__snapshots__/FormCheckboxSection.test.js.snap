// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCheckboxSection render the UI as expected 1`] = `
<div
  className="FormCheckboxSection"
>
  <div
    className="empty"
  />
  <div
    className="FormCheckboxSection__options-container"
  />
</div>
`;
